c/tskit/tables.h

```diff
@@ -2409,6 +2409,24 @@ completes.
 int tsk_table_collection_simplify(tsk_table_collection_t *self, tsk_id_t *samples,
     tsk_size_t num_samples, tsk_flags_t options, tsk_id_t *node_map);
 
+/**
+@brief Subsets a table collection by a set of nodes.
+
+@rst
+Reduces the table collection, retaining only the nodes given in ``nodes``,
```

Comment:
This is a bit terser than the Python docs. C programmers will want precision too. I think we want something a more like the version we have on the python TreeSequence documentation.